Philip Chang // iP PR

philip1304 opened the pull request:
plopBot iP // PR

philip1304 committed:
TEST123

philip1304 committed:
plopBot - Level 01: Echo // Implemented

philip1304 committed:
plopBot - Level 02: Add, List // Implemented

philip1304 committed:
plopBot - Level 03: Mark & Unmark // Implemented

philip1304 committed:
plopBot - Level 04: ToDos, Events, Deadlines // Implemented

philip1304 committed:
plopBot - Level 5: Handle Errors // Implemented

philip1304 committed:
plopBot - Level 6: Delete // Implemented

philip1304 committed:
plopBot - Level 07: Save // W.I.P

philip1304 committed:
plopBot - Level 07: Save // W.I.P + .jar

thomasjlalba commented on src/main/java/PlopBot.java:
```diff
@@ -0,0 +1,314 @@
+        String command = parts[0].toLowerCase();
+        String details = parts.length > 1 ? parts[1] : "";
+
+        switch (command) {
```
For your switch statements, you could follow the following style set by the couse in this [link](https://se-education.org/guides/conventions/java/basic.html#layout)!

thomasjlalba commented on src/main/java/PlopBot.java:
```diff
@@ -0,0 +1,314 @@
+        }
+        catch (Exception e) {
+        }
+        System.out.println("");
```
Maybe you couild indicate a meaningful message for exceptions thrown?

thomasjlalba commented on src/main/java/PlopBot.java:
```diff
@@ -0,0 +1,314 @@
+            catch (IllegalArgumentException ex) {
+                switch (dateString.toLowerCase()) {
+                    case "today":
+                        return now;
```
Do include break statements, and indicate if its an explicit fallthrough

thomasjlalba commented on src/main/java/PlopBot.java:
```diff
@@ -0,0 +1,314 @@
+            String[] parts = dateTimeString.split(" ", 2);
+
+            if (parts.length == 2) {
+                LocalDate date = parseDateString(parts[0]);
```
Maybe it would be good to refactor `parts[0]` to indicate what it means

thomasjlalba commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,131 @@
+        this.taskName = taskName;
+        this.type = TaskType.TODO;
+
+        //this.isDone = false;
```
Do remove unnecessary lines!

thomasjlalba commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,131 @@
+    public String getTypeIcon() {
+        switch (type) {
+            case TODO:
+                return "T";
```
Maybe you can avoid magic strings and store the strings!

thomasjlalba reviewed (commented):
Just some formatting errors and some coding styles to follow for this course! Good work!

philip1304 committed:
plopBot - Level 7: Save // Implemented

philip1304 committed:
plopBot - Level 9: Find // Implemented

philip1304 committed:
plopBot - A-JavaDoc // Implemented & Tidied Up

philip1304 committed:
plopBot - A-UserGuide // Implemented & Tidied Up

philip1304 committed:
plopBot - A-JavaDoc // Implemented & Tidied Up

philip1304 committed:
plopBot - A-JavaDoc // Implemented & Tidied Up

philip1304 committed:
plopBot - Final Jar File // Completed

philip1304 committed:
plopBot - A-JavaDoc // Tidied Up

philip1304 committed:
ip // Final JAR Release

ip // Final JAR Release

philip1304 committed:
PlopBot // Updated

Notes:
- Added abstraction & documentation // Parser.java
- Added abstraction & documentation, fixed errors that would cause the program to exit // PlopBot.java
- Added documentation // Storage.java
- Added documentation, cleaned up code // Task.java
- Added documentation // TaskList.java
- Added documentation // Ui.java
